tasks/mesos.yml

```diff
@@ -41,88 +25,98 @@
+  template: src=conf-mesos.j2 dest=/etc/default/mesos
+  notify:
+  - Restart mesos-master service
+  - Cleanup meta/slaves/latest
+  - Restart mesos-slave service
+
+- block:
+  - name: Mesos Master config file
+    template: src=conf-mesos-master.j2 dest=/etc/default/mesos-master mode=0644 owner=root group=root
+    notify:
+    - Restart mesos-master service
+
+  - name: Mesos master upstart script
+    template: src=init-mesos-master.j2 dest=/etc/init/mesos-master.conf
+    when: upstart_check.stat.exists
+    notify:
+    - Restart mesos-master service
+
+  - name: Mesos master systemd script
+    template: src=mesos-master.service.j2 dest=/etc/systemd/system/mesos-master.service
+    when: systemd_check.stat.exists
+    notify:
+    - Reload systemd
+    - Restart mesos-master service
+
+  - name: Enable/Disable mesos-master service
+    service:
+      name: mesos-master
+      enabled: yes
+    when: >
+      "master" in mesos_install_mode
+
+  - debug:
+      msg: |
+        Forcing restart: {{ force_restart|bool }}
+    notify:
+    - Restart mesos-master service
+    when: force_restart|bool
+
+  - name: Flush handlers
+    meta: flush_handlers
+
+  - name: Start mesos-master service if it has not been started yet
+    service:
+      name: mesos-master
+      state: started
   when: >
     "master" in mesos_install_mode
 
-- name: Cleanup meta/slaves/latest
-  file: path="{{mesos_work_dir}}/meta/slaves/latest" state=absent
-  when: >
-    "slave" in mesos_install_mode
-    and (mesos_slave_config_updated or force_restart|bool)
-    and meta_cleanup|bool
-
-- name: Control status of mesos-slave service
-  service:
-    name: mesos-slave
-    state: "{% if mesos_slave_config_updated or force_restart|bool %}re{% endif %}started"
+- block:
+  - name: Mesos Slave config file
+    template: src=conf-mesos-slave.j2 dest=/etc/default/mesos-slave mode=0644 owner=root group=root
+    notify:
+    - Cleanup meta/slaves/latest
+    - Restart mesos-slave service
+
+  - name: Mesos slave upstart script
+    template: src=init-mesos-slave.j2 dest=/etc/init/mesos-slave.conf
+    when: upstart_check.stat.exists
+    notify:
+    - Cleanup meta/slaves/latest
+    - Restart mesos-slave service
+
+  - name: Mesos slave systemd script
+    template: src=mesos-slave.service.j2 dest=/etc/systemd/system/mesos-slave.service
+    when: systemd_check.stat.exists
+    notify:
+    - Reload systemd
+    - Cleanup meta/slaves/latest
+    - Restart mesos-slave service
+
+  - name: Enable/Disable mesos-slave service
+    service:
+      name: mesos-slave
+      enabled: yes
+
+  - debug:
+      msg: |
+        Forcing restart: {{ force_restart|bool }}
+    notify:
+    - Cleanup meta/slaves/latest
+    - Restart mesos-slave service
+    when: force_restart|bool
+
+  - name: Flush handlers
+    meta: flush_handlers
+
+  - name: Start mesos-slave service if it has not been started yet
+    service:
+      name: mesos-slave
+      state: started
```

Comment:
IMO *not really* required to ensure it's started (like explained, it only applies to special case), but can live with it. Just want to see the opinion of the upstream repos maintainer also.

what about the same for the 'master' ?